src/lib/txn.js

```diff
@@ -92,7 +92,7 @@ const signTransaction = async ({
   } else if (walletType === WALLET_TYPES.TREZOR) {
     await trezorSignTransaction(stx, walletHdPath);
   } else if (walletType === WALLET_TYPES.METAMASK) {
-    return metamaskSignTransaction(stx, wallet.address);
+    return metamaskSignTransaction(utx, wallet.address);
```

Comment:
Can you explain this change? This cascades down into the changes in `metamaskSignTransaction`, which seem to give us less guarantees about the format of the data we pass to Metamask. But maybe I'm wrong and/or Metamask doesn't care?

Reply:
by not putting it in a Tx object we preserve some properties which allows us to use web3 instead of manually sending the RPC request. This means that the error being thrown is consistently an `Error` object and not some other weird jankery.